### ISSUE TITLE
Central Authentication of Ops Users onto EC2 instances

### ISSUE THREAD

**Issue by cassidti** [UKHomeOffice/dq-aws-migration#70]
Need a solution/service for EC2 (win + linux) Authentication.

Simple AD service is not available currently within the London AWS Region.

### Acceptance Criteria
- [x] Identify solution options
- [x] AD provisioned
- [x] Redhat auth via AD creds
- [x] Windows auth via AD creds
- [x] Document any required networking/security group details for connectivity requirements
- [x] Test cross account AD usage (AD in one account, ec2 instances in another to demonstrate ops vs prod etc)
- [x] ~~Update connectivity tester AMI to allow auth via AD creds~~
- [x] ad module https://github.com/ukhomeoffice/dq-tf-ad
- [x] representative redhat + windows boxes that auth in ops
- [x] representative redhat + windows boxes that auth in apps

**Comment by cassidti** [comment]
AWS Directory Service for Microsoft Active Directory (Standard Edition) is available in eu-west-2

**Comment by chrisns** [comment]
http://docs.aws.amazon.com/directoryservice/latest/admin-guide/join_a_directory.html

**Comment by chrisns** [comment]
Blocked waiting on approval for use of Directory Services or access to CTI

**Comment by chrisns** [comment]
could potentially use http://mutonufoai.github.io/pgina/ and write a SAML or basic HTTP auth to use with Keycloak

**Comment by chrisns** [comment]
... or auth to AWS with
```
username: AWS_ACCESS_KEY_ID
password: AWS_ACCESS_SECRET
```

**Comment by chrisns** [comment]
blocked by #136

**Comment by cassidti** [comment]
#145 has been closed as it covered the same scope as criteria representative redhat + windows boxes that auth in ops/app to be completed

**Comment by david-wirelab** [comment]
TF state still has got SSM association `                "ad_aws_ssm_document_name": {
                    "sensitive": false,
                    "type": "string",
                    "value": "ssm_doc_EWx8K2JI"
                }` hence blocking AD deployment, S3 buckets and connectivity tester update.

**Comment by david-wirelab** [comment]
Issue with TF state resolved. Removed Blocker flag.

**Comment by david-wirelab** [comment]
Removed task for AD pwd in KMS as there is a separate issue for the same https://github.com/UKHomeOffice/dq-aws-transition/issues/147
